src/io/flutter/perf/FlutterWidgetPerf.java

```diff
@@ -20,7 +20,6 @@
 import com.intellij.openapi.util.TextRange;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.util.ui.EdtInvocationManager;
-import gnu.trove.TIntObjectHashMap;
```

Comment:
I believe this is a performance optimization (`TIntObjectHashMap` - optimized when the keys of your map are an int). I don't know if it's actually required in the places of the code where we're using `TIntObjectHashMap`.

cc @jacob314 for this use site

Have you looked to see if there's an equivalent in the IntelliJ codebase, or our other deps?

Reply:
What frequency of operations are we expecting that the hashing algorithm used might be a performance hit?

Reply:
Looking at this, I believe we may get better worst-case performance on java.util.HashMap than gnu.trove.THashMap.

Looking at the hashing strategy used in [TIntObjectHashMap](https://github.com/palantir/trove-3.0.3/blob/master/generated-src/gnu/trove/impl/hash/TIntHash.java#L256), it reads like a simple double-hashing algorithm.

Java HashMaps use [balanced tree buckets](http://openjdk.java.net/jeps/180) in degenerate hashing cases.  This approach reduces the number of hash collisions to handle and gives asymptotically better big-O performance.  Both, naturally, have amortized O(1) performance.  However, the tree bucketing algorithm gives un-amortized O(log(n)) put and get operations, while a double-hash gives un-amortized O(n).